prototype_source/fx_numeric_suite_tutorial.py

```diff
@@ -0,0 +1,231 @@
+# -*- coding: utf-8 -*-
+"""
+PyTorch FX Numeric Suite Core APIs Tutorial
+==============================
+
+Introduction
+------------
+
+Quantization is good when it works, but it’s difficult to know what's wrong
+when it doesn't satisfy the accuracy we expect. Debugging the accuracy issue
+of quantization is not easy and time consuming.
+
+One important step of debugging is to measure the statistics of the float model
+and its corresponding quantized model to know where are they differ most.
+We built a suite of numeric tools called PyTorch FX Numeric Suite Core APIs in
+PyTorch quantization to enable the measurement of the statistics between
+quantized module and float module to support quantization debugging efforts.
+Even for the quantized model with good accuracy, PyTorch FX Numeric Suite Core
+APIs can still be used as the profiling tool to better understand the
+quantization error within the model and provide the guidance for further
+optimization.
+
+PyTorch FX Numeric Suite Core APIs currently supports models quantized through
+both static quantization and dynamic quantization with unified APIs.
+
+In this tutorial we will use MobileNetV2 as an example to show how to use
+PyTorch FX Numeric Suite Core APIs to measure the statistics between static
+quantized model and float model.
+
+Setup
+^^^^^^
+We’ll start by doing the necessary imports:
+"""
+
+##############################################################################
+
+# Imports and util functions
+
+from __future__ import print_function, division, absolute_import
+import copy
+import torch
+import torchvision
+import torch.quantization
+import torch.ao.ns._numeric_suite_fx as ns
+import torch.quantization.quantize_fx as quantize_fx
+
+import matplotlib.pyplot as plt
+from tabulate import tabulate
+
+torch.manual_seed(0)
+plt.style.use('seaborn-whitegrid')
+
+
+# a simple line graph
+def plot(xdata, ydata, xlabel, ylabel, title):
+    _ = plt.figure(figsize=(10, 5), dpi=100)
+    plt.xlabel(xlabel)
+    plt.ylabel(ylabel)
+    plt.title(title)
+    ax = plt.axes()
+    ax.plot(xdata, ydata)
+    plt.show()
+
+##############################################################################
+# Then we load the pretrained float MobileNetV2 model, and quantize it.
+
+
+# create float model
+mobilenetv2_float = torchvision.models.quantization.mobilenet_v2(
+    pretrained=True, quantize=False).eval()
+
+# create quantized model
+qconfig_dict = {
+    '': torch.quantization.get_default_qconfig('fbgemm'),
+    # adjust the qconfig to make the results more interesting to explore
+    'module_name': [
+        # turn off quantization for the first couple of layers
+        ('features.0', None),
+        ('features.1', None),
+        # use MinMaxObserver for `features.17`, this should lead to worse
+        # weight SQNR
+        ('features.17', torch.quantization.default_qconfig),
+    ]
+}
+# Note: quantization APIs are inplace, so we save a copy of the float model for
```

Comment:
```suggestion
# Note: quantization APIs are in place, so we save a copy of the float model for
```

Reply:
keeping it as is, lmk if this needs to change